### PR TITLE
Refactor sidebar and docs

### PR DESCRIPTION
**Pull request by MikaelSmith**
Extract header and sidebar for modularity and to make them easier to read. Move more of docs navigation to the sidebar.

Signed-off-by: Michael Smith <michael.smith@puppet.com>

### CODE REVIEW

**Comment by MikaelSmith** [comment]
This you definitely need to preview with Jekyll.

File: docs/docs.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+title: Docs
+---
```
Comment (MikaelSmith) [inline docs/docs.md]: Not sure what content we should put here. Most basic would be a list reflecting the links in the sidebar.

Comment (ekinanp) [reply docs/docs.md]: Maybe? Do we need to put something here?

**Comment by ekinanp** [comment]
Kind of leaning towards flattening out "Concepts".

**Comment by MikaelSmith** [comment]
They all felt related to me. Have an idea how you'd organize that?

**Comment by ekinanp** [comment]
I was thinking just put it top-level. So under `Docs` you'd have a link to `CName`, link to `Attributes`, link to `Actions`.

**Comment by MikaelSmith** [comment]
Maybe Concepts needs a preamble. I still feel like actions/cname/attributes are best understood as a set of concepts rather than distinct topics.

**Comment by ekinanp** [comment]
I guess. I was kind of thinking of it as a user saying "Ok.. What were the supported actions?" => "Ah, I can see that as "Actions" under Docs", etc and that being the common case. But I'm not against what you have either, so up to you.

**Comment by MikaelSmith** [comment]
I lean conservative for now, and we can change it later once we get more feedback.

**Review by ekinanp — APPROVED** [review]

**Pull request merged**